lib/ansible/modules/cloud/vmware/vmware_host_config_manager.py

```diff
@@ -164,19 +165,33 @@ def set_host_configuration_facts(self):
                     if option_value != host_facts[option_key]['value']:
                         change_option_list.append(vim.option.OptionValue(key=option_key, value=option_value))
                         changed = True
+                        changed_list.append(option_key)
                 else:  # Don't silently drop unknown options. This prevents typos from falling through the cracks.
                     self.module.fail_json(msg="Unknown option %s" % option_key)
             if changed:
-                try:
-                    option_manager.UpdateOptions(changedValue=change_option_list)
-                except (vmodl.fault.SystemError, vmodl.fault.InvalidArgument) as e:
-                    self.module.fail_json(msg="Failed to update option/s as one or more OptionValue "
-                                              "contains an invalid value: %s" % to_native(e.msg))
-                except vim.fault.InvalidName as e:
-                    self.module.fail_json(msg="Failed to update option/s as one or more OptionValue "
-                                              "objects refers to a non-existent option : %s" % to_native(e.msg))
-
-        self.module.exit_json(changed=changed)
+                if self.module.check_mode:
+                    changed_prefix = ' would be changed.'
+                else:
+                    changed_prefix = ' changed.'
+                if len(changed_list) > 2:
+                    message = ', '.join(changed_list[:-1]) + ', and ' + str(changed_list[-1]) + changed_prefix
+                elif len(changed_list) == 2:
+                    message = ' and '.join(changed_list) + changed_prefix
+                elif len(changed_list) == 1:
+                    message = changed_list[0] + changed_prefix
```

Comment:
small nit pick, but you can ignore as well. Add changed_prefix at end of if as it is common in all 3 if blocks.

Reply:
I might remove `changed_prefix` since you want the same messages for check mode and non-check mode

Reply:
Ok. I changed it. thx.